frontend/src/pages/Profile.js

```diff
@@ -29,44 +32,144 @@ export default class Place extends Component {
         Authorization: token
       }
     })
-    this.setState({ token: token, user: user.data, posts: postsData.data })
+
+    for(let i = 0; i < postsData.data.lenght; i++) {
+      postsData.data[i].showPostModal = false
+    }
+
+    this.setState({
+      token: token,
+      user: user.data,
+      posts: postsData.data,
+      showPostModal: false,
+      showProfileModal: false
+    })
   }
 
-  render() {
-    const { user, posts } = this.state
+  handleShowPostModal(index) {
+    const { posts } = this.state
+    posts[index].showPostModal = true
+    posts.forEach((post, postIndex) => {
+      if (postIndex === index) post.showPostModal = true
+      else post.showPostModal = false
+    })
+    this.setState({ posts: posts, showProfileModal: false })
+  }
 
+  handleShowProfileModal() {
+    const { posts } = this.state
+    for(let i = 0; i < posts.length; i++){
+      posts[i].showPostModal = false
+    }    
+    this.setState({ showProfileModal: true, posts: posts })
+  }
+
+  async handleEditProfile(username, avatar_url) {
+    let token = localStorage.getItem('token')
+    let user = await API.put('/users', {
+      username: username,
+      avatar_url: avatar_url
+    }, {
+      headers: {
+        Authorization: token
+      }
+    })
+    if (user.data) {
+      notification.success({
+        message: 'SUCCESS',
+        description: `UPDATED SUCCESSFULL`
+      })
+    } else {
+      notification.error({
+        message: 'ERROR',
+        description: `SERVER DOWN`
+      })
+    }
+    window.location.reload()
+  }
+
+  async handleDelete(id) {
+    const { posts } = this.state
+    let token = localStorage.getItem('token')
+    let deleted = await API.delete(`/places/3/posts`, {
+      id: id
+    }, {
+      headers: {
+        Authorization: token
+      }
+    })
```

Comment:
Method `delete` của axios cơ bản sẽ hơi khác so với `post`, `put`, `patch` (https://github.com/axios/axios/issues/897#issuecomment-343715381)
Try this:
```suggestion
    let deleted = await API.delete(`/places/2/posts`,
        {
          data: {
            id: id
          },
          headers: {
            Authorization: token
          },
        },
      )
```

Reply:
Ờ nhỉ, quên mất cái này đó